tests/ui/coroutine/static-mut-reference-across-yield.rs

```diff
@@ -10,7 +10,7 @@ fn main() {
     unsafe {
         let gen_index = #[coroutine]
         static || {
-            let u = A[{
+            let u = A[{ //~ WARN creating a reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.